test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +282,91 @@ var _ = Describe("Garbage collection for dependent resources", func() {
+			secretName    = "dependentsecret"
+		)
+
+		var (
+			owner        v1alpha1.ClusterServiceVersion
+			fetchedOwner *v1alpha1.ClusterServiceVersion
+			propagation  metav1.DeletionPropagation
+			options      metav1.DeleteOptions
+			configMap    *corev1.ConfigMap
+			secret       *corev1.Secret
+		)
+
+		BeforeEach(func() {
+			var err error
+			owner = newCSV("owner", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("test", nil, nil))
+			// create all owners
+			fetchedOwner, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(context.TODO(), &owner, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// wait for csv install to complete
+			Eventually(func() bool {
+				csv, _ := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), fetchedOwner.GetName(), metav1.GetOptions{})
+				return csv.Status.Phase == v1alpha1.CSVPhaseSucceeded
+			}).Should(BeTrue())
+
+			cfg := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configmapName,
+					Namespace: testNamespace,
+				},
+				Data: map[string]string{"hello": "there"},
+			}
+
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: testNamespace,
+				},
+				Data: map[string][]byte{"name": []byte("Alice")},
+			}
+
+			// set ownerrefs to objects
+			ownerutil.AddOwner(cfg, fetchedOwner, true, false)
+			ownerutil.AddOwner(s, fetchedOwner, true, false)
+
+			// Create a configmap
+			configMap, err = kubeClient.CreateConfigMap(cfg)
+			Expect(err).NotTo(HaveOccurred())
+
+			// Create a secret
+			secret, err = kubeClient.CreateSecret(s)
+			Expect(err).NotTo(HaveOccurred())
+
+			propagation = metav1.DeletePropagationForeground
+			options = metav1.DeleteOptions{PropagationPolicy: &propagation}
+		})
+
+		When("CSV is deleted", func() {
+
+			BeforeEach(func() {
+				// Delete CSV
+				err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Delete(context.TODO(), fetchedOwner.GetName(), options)
+				Expect(err).To(BeNil())
+
+				// wait for deletion
+				Eventually(func() bool {
+					_, err := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), fetchedOwner.GetName(), metav1.GetOptions{})
+					return k8serrors.IsNotFound(err)
+				}).Should(BeTrue())
+			})
+
+			It("should delete the associated configmap and secret", func() {
+				_, err := kubeClient.GetConfigMap(testNamespace, configMap.GetName())
+				Expect(err).To(HaveOccurred())
+				Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+				ctx.Ctx().Logf("configmap successfully garbage collected after csv deleted")
+
+				_, err = kubeClient.GetSecret(testNamespace, secret.GetName())
+				Expect(err).To(HaveOccurred())
+				Expect(k8serrors.IsNotFound(err)).To(BeTrue())
+				ctx.Ctx().Logf("secret successfully garbage collected after csv deleted")
```

Comment:
We may want to put this in an `Eventually` block.

Reply:
Originally I did have these in `Eventually` blocks, but I realized checking it this way is actually more precise. Since we are using the foreground deletion option when deleting the CSV, that ensures that dependent objects (the secret and configmap) are GC'd before the CSV itself. Therefore once the CSV is gone we should be 100% sure that the dependent objects are gone as well. Checking it this way confirms that to be the case

Reply:
Ah great point!